pkg/api/kubegraph/analysis/hpa.go

```diff
@@ -0,0 +1,155 @@
+package analysis
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	graphapi "github.com/gonum/graph"
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	"github.com/openshift/origin/pkg/api/kubegraph"
+	kubenodes "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+	deploygraph "github.com/openshift/origin/pkg/deploy/graph"
+	deploynodes "github.com/openshift/origin/pkg/deploy/graph/nodes"
+)
+
+const (
+	HPAMissingCPUTargetError      = "HPAMissingCPUTarget"
+	HPAOverlappingScaleRefWarning = "HPAOverlappingScaleRef"
+)
+
+// FindHPASpecsMissingCPUTargets scans the graph in search of HorizontalPodAutoscalers that are missing a CPU utilization target.
+// As of right now, the only metric that HPAs can use to scale pods is the CPU utilization, so if a HPA is missing this target it
+// is effectively useless.
+func FindHPASpecsMissingCPUTargets(graph osgraph.Graph, namer osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastNode := range graph.NodesByKind(kubenodes.HorizontalPodAutoscalerNodeKind) {
+		node := uncastNode.(*kubenodes.HorizontalPodAutoscalerNode)
+
+		hpa, ok := node.Object().(*extensions.HorizontalPodAutoscaler)
+		if !ok {
+			continue
+		}
+
+		if hpa.Spec.CPUUtilization == nil {
+			markers = append(markers, osgraph.Marker{
+				Node:     node,
+				Severity: osgraph.ErrorSeverity,
+				Key:      HPAMissingCPUTargetError,
+				Message:  fmt.Sprintf("%s is missing a CPU utilization target", namer.ResourceName(node)),
```

Comment:
> We can always propose to them to patch in the default. @deads2k WDYT?

That seems reasonable to me.  @DirectXMan12 is there ever a reason why that would be bad advice?